### ISSUE TITLE
Timeout error even with no timeout used on bblfshd

### ISSUE THREAD

**Issue by ajnavarro** [bblfsh/javascript-driver#43]
Removing the context timeout using the go bblfsh client, I'm still having the following error when parsing javascript files:

```
Processing file docs/js/vendor/jquery-1.8.0.min.js
rpc error: code = DeadlineExceeded desc = context deadline exceeded
```

**Comment by dennwc** [comment]
`bblfshd` always forces a timeout if you don't set one. You can try setting a higher timeout on your request, so `bblfshd` will use your value instead of the default.

**Comment by ajnavarro** [comment]
Even starting the client in this way?:

```
client, err := bblfsh.NewClientContext(context.TODO(), endpoint)
```

I didn't have problems with other drivers like java.

**Comment by dennwc** [comment]
Starting is okay, but the request should have a timeout set, or it will be set by `bblfshd`.

**Comment by bzz** [comment]
@ajnavarro the fact the it's only JS may have to do with your example using a big minified file that takes time to parse.

On the broader scope - it was a bit surprising for me as a user as well, but what I learned is that in order to set different timeout one needs to: 

```go
ctx, cancel := context.WithTimeout(context.Background(), 50*time.Second)
defer cancel() 
resp, err := client.NewParseRequest().Context(ctx).ReadFile(srcFile).Do()
//...
```

Does this solve your use case, and thus we can close this? 

Or do you rather suggest making a change, so any request will respect client's context (that could be eventually just overridden by the `.Context()`) so:

```go
client, err := bblfsh.NewClientContext(ctx, endpoint)
resp, err := client.NewParseRequest().ReadFile(srcFile).Do()
```

is equivalent to current

```go
resp, err := client.NewParseRequest().Context(ctx).ReadFile(srcFile).Do()
```

?

I personally think the UX (or better DX - Developer Experience :) ) of the client would be improved with such change and would be happy to ask @smola to move this issue to go-client and add it, if we all agree and there are no objections.

@dennwc @creachadair  WDYT?

**Comment by dennwc** [comment]
@bzz Binding the request context for the whole client lifetime is not the best option. If you don't want a per-request timeout, you can just omit `.Context()` call.

**Comment by creachadair** [comment]
Yeah, generally timeouts should only be introduced with care and deliberation. It's too easy to mask other problems under a chain of timeouts, and it's really hard to debug. And they should definitely be _request_ scoped, not _client_ scoped, if at all.

**Comment by ajnavarro** [comment]
@bzz we are setting the context also in the NewParseRequest, but even with that, I was not able to parse jquery file, always the same error after 5 secs.

**Comment by bzz** [comment]
@dennwc @creachadair 
Thank you for feedback and explanations!

> Binding the request context for the whole client lifetime is not the best option

I agree. 

May be I misunderstood something, but I was looking at the proposal above as a way to override _already existing, default timeout_ (not introducing another one) of [5 sec](https://github.com/bblfsh/client-go/blob/master/client.go#L39) for ALL subsequent requests done though this client.

Would not this be the case?


@ajnavarro Did you try increasing the timeout?

I just tried

```bash
wget http://code.jquery.com/jquery-1.8.0.min.js

cat << 'EOF' > client.go
package main

import (
	"fmt"

	"gopkg.in/bblfsh/client-go.v3"
	"gopkg.in/bblfsh/sdk.v2/uast/yaml"
)

func main() {
	client, err := bblfsh.NewClient("localhost:9432")
	if err != nil {
		panic(err)
	}

	res, _, err := client.NewParseRequest().ReadFile("jquery-1.8.0.min.js").UAST()
	if err != nil {
		panic(err)
	}

	data, err := uastyml.Marshal(res)
	if err != nil {
		panic(err)
	}
	fmt.Println(string(data))
}
EOF

go get -v ./...
time go run ./clinet.go
```

and it fails with `panic: rpc error: code = DeadlineExceeded desc = context deadline exceeded` in 6s, but if you try adding a context to request with timeout 30sec as noted above, `time go run ./client.go | wc -l` returns in ~10sec.

Could you please try and let me know if that solves the issue? 

I'm using `bblfsh/bblfshd:v2.9.2` with `bblfsh/javascript-driver:latest` which is `v2.4.1` (until y'day).
